rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/typesystem.kt

```diff
@@ -242,6 +242,19 @@ data class NumberType(val entireDigits: Int, val decimalDigits: Int, val rpgType
             return false
         }
     }
+
+    override fun equals(other: Any?): Boolean {
+        return if (other is NumberType) {
+            val resultDigits = this.entireDigits == other.entireDigits && this.decimalDigits == other.decimalDigits
+            if (rpgType?.isNotBlank()!! && other.rpgType?.isNotEmpty()!!) {
```

Comment:
I don't understand why in order to test `rpgType`. `this.rpgType` must be not blank while `other.rpgType` must be not empty.